stargate/pkg/auth/client.go

```diff
@@ -0,0 +1,96 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type TableBasedTokenProvider struct {
+	client   *client
+	username string
+	password string
+}
+
+type client struct {
+	serviceURL string
+	httpClient *http.Client
+}
+
+type AuthResponse struct {
+	AuthToken string `json:"authToken"`
+}
+
+type AuthRequest struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+func NewTableBasedTokenProvider(serviceURL, username, password string) TableBasedTokenProvider {
+	return TableBasedTokenProvider{
+		client:   getClient(serviceURL),
+		username: username,
+		password: password,
+	}
+}
+
+func (t TableBasedTokenProvider) GetToken() (string, error) {
+	AuthReq := AuthRequest{
```

Comment:
This seems pretty expensive to run on every query/batch request. Could it cache?

Reply:
I'm wonder if there shouldn't be some way for passing in a `Context` here to handle cancelling (or handling timeouts) for the http request.

Reply:
Cached